warehouse/query-core/src/main/java/datawave/query/data/parsers/FieldIndexKey.java

```diff
@@ -0,0 +1,106 @@
+package datawave.query.data.parsers;
```

Comment:
I like this - especially how it avoids doing unnecessary or redundant work. Consider how this will degrade in an state where the parse of the key fails in either `traverseColumnQualifier` or calls to any of the `get` methods. You could add a 'parseError' flag, for instance, that gets set when an error is encountered and used to guide the response of the various `get` methods. 

Depending on the scope of reuse, it also might be helpful to consider implementing a 'reset' method or 'clear' method to call when we are done so this doesn't hold onto a reference to the last key parsed indefinitely.

Reply:
Spent some time reworking the whole class. 
- the `traverseColumnQualifier()` method will only be called once
- getters will return null values if the backing array is invalid
- added a `clearState()` method